src/lib/DepositController.js

```diff
@@ -9,29 +9,38 @@
 // Defines what happens when a button is clicked.
 
 import { setFormErrorsFromResponse } from './state/actions';
-import { PUBLISH_SUCCESS, SAVE_SUCCESS } from './state/types';
+import {
+  CREATE_DEPOSIT_SUCCESS,
+  PUBLISH_SUCCESS,
+  SAVE_SUCCESS,
+} from './state/types';
 
 export class DepositController {
-  constructor(apiClient) {
+  constructor(apiClient, fileUploader) {
     this.apiClient = apiClient;
+    this.fileUploader = fileUploader;
   }
 
   exists(record) {
-    return record.id ? true : false;
+    return record.id || record.pid ? true : false;
   }
 
   validate(record) {
     console.log('Validate record', record);
   }
 
-  async save_draft(record, { formik, store }) {
+  async saveDraft(record, { formik, store }) {
     // Saves a draft of the record
     const recordSerializer = store.config.recordSerializer;
     let payload = recordSerializer.serialize(record);
```

Comment:
```suggestion
    const payload = recordSerializer.serialize(record);
```

Reply:
It's a `let` because in case of creation I assign the request `response.data` to it!